who_is_active.sql

```diff
@@ -3679,7 +3679,50 @@ BEGIN;
 			@statement_start_offset INT,
 			@statement_end_offset INT,
 			@start_time DATETIME,
-			@database_name sysname;
+			@database_name SYSNAME,
+			@server_name SYSNAME = @@SERVERNAME,
+			@skip_databases_sql NVARCHAR(MAX) = N'
+				SELECT
+					d.name
+				FROM sys.databases AS d
+				INNER JOIN sys.availability_replicas AS r
+					ON d.replica_id = r.replica_id
+				WHERE NOT EXISTS
+					  (
+						  SELECT
+							  1/0
+						  FROM sys.dm_hadr_availability_group_states AS s
+						  WHERE s.primary_replica = r.replica_server_name
+					  )
+				AND r.secondary_role_allow_connections_desc = ''READ_ONLY''
+				AND r.replica_server_name = @server_name

```

Comment:
Is there a reason to go dynamic here? Why not just use @@SERVERNAME hardcoded in a non-dynamic query?

Reply:
I thought that I remembered this throwing an error if the DMV doesn’t exist (pre-2012).

Reply:
Oh, like an object resolution thing? That makes sense.

Reply:
Yep, exactly. I can test tomorrow if you really don’t like the dynamic stuff here.